src/loaders/ObjectLoader.js

```diff
@@ -466,21 +467,31 @@ Object.assign( ObjectLoader.prototype, {
 
 				if ( data.type === 'MultiMaterial' ) {
 
-					// Deprecated
-
+                    			// Deprecated
 					var array = [];
-
 					for ( var j = 0; j < data.materials.length; j ++ ) {
 
-						array.push( loader.parse( data.materials[ j ] ) );
+						if ( cache[ data.materials[ j ].uuid ] ) {
+
+							array.push( cache[ data.materials[ j ].uuid ] );
+
+						} else {
+
+							var mat = loader.parse( data.materials[ j ] );
+							cache[ data.materials[ j ].uuid ] = mat;
+							array.push( mat );
+
+						}
 
 					}
 
 					materials[ data.uuid ] = array;
+					cache[ data.uuid ] = array;
```

Comment:
Is this one needed?

Reply:
Maybe not

Reply:
Would you like to do another PR removing it? Just to keep the code clear.

Reply:
sure

Reply:
Actually, I'll clean it up on my end 👌

Reply:
https://github.com/mrdoob/three.js/commit/f8dc0a9335a18eda38bf028e54cca562ea11e0d5

Reply:
<img width="654" alt="screen shot 2018-10-31 at 10 39 05 am" src="https://user-images.githubusercontent.com/97088/47795654-478a1800-dcf9-11e8-8109-23594953798c.png">